enclave-tls/src/crypto_wrappers/internal/etls_crypto_wrapper_select.c

```diff
@@ -32,6 +33,23 @@ enclave_tls_err_t etls_crypto_wrapper_select(etls_core_context_t *ctx,
 		if (type && strcmp(type, crypto_wrappers_ctx[i]->opts->type))
 			continue;
 
+		if (crypto_wrappers_ctx[i]->opts->flags) {
+                        bool sgx_supported = (is_sgx2_supported() || is_sgx1_supported());
+                        if (!sgx_supported) {
+                                ETLS_ERR("Selecting crypto wrapper while lacking sgx \
+                                 capability\n");
+                                return -ENCLAVE_TLS_ERR_INIT;
+                        }
+                } else {
+                        bool sgx_device_exists = (is_dcap_oot_kernel_driver() ||
+                        is_legacy_oot_kernel_driver() || is_in_tree_kernel_driver());
+                        if (!sgx_device_exists) {
+                                ETLS_ERR("Selecting crypto wrapper while sgx \
+                                                device file doesn't exist\n");
+                                return -ENCLAVE_TLS_ERR_INIT;
+                        }
+                }
+
```

Comment:
why crypto_wrappers_ctx[i]->opts->flags is a mutex with sgx kernel driver? sgx kernel driver and sgx capablity should be on the same page.